src/service/storage-impl.js

```diff
@@ -223,12 +224,19 @@ export class Store {
 
   /**
    * @param {string} name
+   * @param {number|undefined} opt_duration
    * @return {*|undefined}
    */
-  get(name) {
+  get(name, opt_duration) {
     // The structure is {key: {v: *, t: time}}
     const item = this.values_[name];
-    return item ? item['v'] : undefined;
+    const timestamp = item ? item['t'] : undefined;
+    const isNotExpired =
+      opt_duration && timestamp != undefined
```

Comment:
Is 0 or negative value a valid opt_duration value? We need to handle them specially if so.

Reply:
They are not valid values, so I don't think we need to handle them specifically.